### PR TITLE
New Event Page Design

### LINKED ISSUE #537: Unable to Scroll After Deleting an Event
After deleting an event in the event manager, the user is unable to scroll up or down on the page.

### PR DESCRIPTION
**Pull request by quinklan**
Resolves #537

### CODE REVIEW

File: api/main_endpoints/models/Event.js [hunk 1]
```diff
@@ -30,9 +30,10 @@ const EventSchema = new Schema(
     eventCategory: {
       type: String
     },
+
```
Comment (evanugarte) [inline api/main_endpoints/models/Event.js:33]:
```suggestion
```

File: api/main_endpoints/models/Event.js [hunk 2]
```diff
@@ -30,9 +30,10 @@ const EventSchema = new Schema(
...
     imageURL: {
       type: String,
-      default: DEFAULT_PHOTO_URL
+      default: 'https://www.freeiconspng.com/uploads/no-image-icon-11.PNG'
```
Comment (evanugarte) [inline api/main_endpoints/models/Event.js:36]: can you keep the `DEFAULT_PHOTO_URL` here?

File: src/Pages/Events/EventCard.js [hunk 3]
```diff
@@ -1,53 +1,65 @@
 import React from 'react';
-import { Container, Row, Col } from 'reactstrap';
+import { Row } from 'reactstrap';
 import { getDateWithSlashes } from '../../APIFunctions/Event';
-import { mapPinSymbol, clockSymbol} from '../Overview/SVG';
+import { mapPinSymbol, clockSymbol } from '../Overview/SVG';
+// import { sourcerepo } from 'googleapis/build/src/apis/sourcerepo';
```
Comment (evanugarte) [inline src/Pages/Events/EventCard.js:5]:
```suggestion
```

File: src/Pages/Events/EventCard.js [hunk 4]
```diff
@@ -1,53 +1,65 @@
...
     title,
-    description,
     eventDate,
     eventLocation,
     startTime,
     endTime,
     handleClick,
-    imageURL
+    imageURL,
+    noImageURL = 'https://www.freeiconspng.com/uploads/no-image-icon-11.PNG',
```
Comment (evanugarte) [inline src/Pages/Events/EventCard.js:16]: it would be better to send `imageURL` as a prop that already handles if it isnt defined or not in the parent component
```suggestion
```

File: src/Pages/Events/EventCard.js [hunk 5]
```diff
@@ -1,53 +1,65 @@
...
+        className='event-card-button'
+        onClick={() => {
+          handleClick();
+        }}
+      >
+        <table>
+          <tbody>
+            <tr>
+              <td>
+                {imageURL !== null ? (
```
Comment (evanugarte) [inline src/Pages/Events/EventCard.js:31]: you can remove this check with my comment above in this file

File: src/Pages/Events/EventCard.js [hunk 6]
```diff
@@ -1,53 +1,65 @@
...
-      </Row>
-    </Container>
+    <div>
+      <button
+        className='event-card-button'
+        onClick={() => {
+          handleClick();
+        }}
+      >
+        <table>
```
Comment (evanugarte) [inline src/Pages/Events/EventCard.js:27]: does each EventCard need its own `table`?

File: src/Pages/Events/EventInfoModal.js [hunk 7]
```diff
@@ -7,38 +7,50 @@ import {
   Row,
   Col,
   ModalFooter,
-  Button
+  Button,
```
Comment (evanugarte) [inline src/Pages/Events/EventInfoModal.js:10]:
```suggestion
  Button
```

File: src/Pages/Events/EventList.js [hunk 8]
```diff
@@ -13,11 +13,11 @@ function AnnouncementList() {
   const modalProps = {
     currentEvent,
     modal,
-    toggle
+    toggle,
```
Comment (evanugarte) [inline src/Pages/Events/EventList.js:16]: you can revert the changes in this file

File: src/Pages/MembershipApplication/GoogleRecaptcha.js [hunk 9]
```diff
@@ -1,6 +1,6 @@
 import React, { Component } from 'react';
 import ReCAPTCHA from 'react-google-recaptcha';
-import { sitekey } from '../../config/config.json';
+import { sitekey } from '../../config/config';
```
Comment (evanugarte) [inline src/Pages/MembershipApplication/GoogleRecaptcha.js:3]: you can revert the changes in this file

File: src/config/config.example.json [hunk 10]
```diff
@@ -1,7 +1,8 @@
...
-  "MAILER_API_URL": "http://localhost:8082",
-  "RPC_API_URL": "http://localhost:8083"
-}
+    "sitekey": "super secret key",
+    "GENERAL_API_URL": "http://localhost:8080",
+    "LOGGING_API_URL": "http://localhost:8081",
+    "MAILER_API_URL": "http://localhost:8082",
+    "RPC_API_URL": "http://localhost:8083"
+  }
+
```
Comment (evanugarte) [inline src/config/config.example.json:8]: you can revert the changes in this file

File: api/main_endpoints/models/Event.js (hunk 2 above)
Comment (quinklan) [reply api/main_endpoints/models/Event.js:35]: Yup I can keep it

File: src/Pages/Events/EventCard.js (hunk 6 above)
Comment (quinklan) [reply src/Pages/Events/EventCard.js:25]: Nope, the event card does not need its own table but it helped me with formatting initially.

File: src/config/config.example.json (hunk 10 above)
Comment (quinklan) [reply src/config/config.example.json:8]: I am a little confused why it says I added these lines of code when this code is exactly the same as the dev branch

File: src/APIFunctions/Event.js [hunk 11]
```diff
@@ -48,7 +48,7 @@ export async function createNewEvent(newEvent, token) {
     startTime: handleMidnightTime(newEvent.startTime),
     endTime: handleMidnightTime(newEvent.endTime),
     eventCategory: newEvent.eventCategory,
-    imageURL: newEvent.imageURL
+    imageURL: handleImageURL(newEvent.imageURL)
```
Comment (evanugarte) [inline src/APIFunctions/Event.js:51]: define `handleImageURL` function above where you use it, like

```js
foo()

function foo() {
  // ...
}
```

valid:
```js
function foo() {
  // ...
}

foo()
```

File: src/Pages/EventManager/EventManager.js [hunk 12]
```diff
@@ -57,15 +57,18 @@ export default function EventManager(props) {
...
           New Event
         </Button>
         {modal && (
           <EventManagerModal
             modal={modal}
             toggle={toggle}
-            handleDelete={event => deleteEvent(event, props.user.token)}
+            handleDelete={event => {
+              deleteEvent(event, props.user.token);
+              window.location.reload();
```
Comment (evanugarte) [inline src/Pages/EventManager/EventManager.js:70]: why do we need to reoad the page? its better if we just remove the event from the array that is rendered in `deleteEvent`

File: src/Pages/EventManager/EventManagerModal.js [hunk 13]
```diff
@@ -51,7 +54,12 @@ function EventManagerModal(props) {
         addon: 'Event Date*',
         type: 'date',
         defaultValue: props.eventDate ? props.eventDate.slice(0, 10) : '',
-        handleChange: e => setEventDate(e.target.value)
+        handleChange: e => {
+          setEventDate(e.target.value);
+          if(eventDate === undefined) {
```
Comment (evanugarte) [inline src/Pages/EventManager/EventManagerModal.js:59]: why do we check this?

File: src/Pages/Events/EventCard.js [hunk 14]
```diff
@@ -16,38 +15,35 @@ function EventCard(props) {
...
+            {'  '}
+            {getDateWithSlashes(eventDate.slice(0, 10))} {startTime} -
+            {' '}
+            {endTime}
           </Row>
-          <Row className='event-title'>{title}</Row>
           <Row className='event-location'>
             {mapPinSymbol()}
-            {eventLocation}
+            {eventLocation.slice(0, 28)}
```
Comment (evanugarte) [inline src/Pages/Events/EventCard.js:42]: is there a reason why we only keep the first 28 characters?

if you absolutely need this, consider limiting the users input for creating an event instead of handling it later here.

File: src/config/config.example.json [hunk 15]
```diff
@@ -4,4 +4,4 @@
   "LOGGING_API_URL": "http://localhost:8081",
   "MAILER_API_URL": "http://localhost:8082",
   "RPC_API_URL": "http://localhost:8083"
-}
+}
```
Comment (evanugarte) [inline src/config/config.example.json:7]: add blank newline

File: package.json [hunk 16]
```diff
@@ -43,6 +43,7 @@
     "passport": "^0.4.1",
     "pdf-lib": "^1.9.0",
     "react": "^16.13.1",
+    "react-bootstrap": "^1.0.1",
```
[the remark this answers is not in the document]
Comment (quinklan) [reply package.json:46]: Removed

File: src/APIFunctions/Event.js (hunk 11 above)
Comment (quinklan) [reply src/APIFunctions/Event.js:51]: Done. Should I do the same for the handleMidnightTime function as well? I did not code this function but it is defined below the function it is used.

Comment (evanugarte) [reply src/APIFunctions/Event.js:51]: First rebase your branch, it may have already been changed

File: src/Pages/EventManager/EventManager.js (hunk 12 above)
Comment (quinklan) [reply src/Pages/EventManager/EventManager.js:70]: If we only delete the rendered event from the array, the administrator loses the ability to scroll up and down the page and would have to refresh the page in order to scroll again.

File: src/Pages/EventManager/EventManagerModal.js (hunk 13 above)
Comment (quinklan) [reply src/Pages/EventManager/EventManagerModal.js:59]: I was attempting to create an alert whenever the user enters an invalid date when creating an event so I left that code in there for testing purposes, but I was never able to successfully resolve that issue.

File: src/Pages/Events/EventCard.js (hunk 14 above)
Comment (quinklan) [reply src/Pages/Events/EventCard.js:42]: Thank you for the suggestion. I just implemented code to limit the users' input instead.

**Comment by jerrylee17** [comment]
image
I get this error

**Comment by jerrylee17** [comment]
@evanugarte Your comments are the only ones blocking this merge LOL

File: src/Pages/Events/EventCard.js [hunk 17]
```diff
@@ -16,38 +15,35 @@ function EventCard(props) {
...
+            {'  '}
+            {getDateWithSlashes(eventDate.slice(0, 10))} {startTime} -
+            {' '}
+            {endTime}
           </Row>
-          <Row className='event-title'>{title}</Row>
           <Row className='event-location'>
             {mapPinSymbol()}
             {eventLocation}
           </Row>
```
Comment (IsitaB) [inline src/Pages/Events/EventCard.js:43]: ooh you intentionally omitted the description

File: src/Pages/Events/EventCard.js [hunk 18]
```diff
@@ -16,38 +15,35 @@ function EventCard(props) {
...
+          </td>
+          <Row className='event-title'>{title}</Row>
           <Row className='event-date'>
-            {clockSymbol()}{'  '}
-            {getDateWithSlashes(eventDate.slice(0, 10))} {startTime} - {endTime}
+            {clockSymbol()}
+            {'  '}
+            {getDateWithSlashes(eventDate.slice(0, 10))} {startTime} -
+            {' '}
+            {endTime}
```
Comment (IsitaB) [inline src/Pages/Events/EventCard.js:38]: Screen Shot 2020-08-23 at 9 02 36 PM
get rid of the red `selected` outline

File: src/APIFunctions/Event.js [hunk 19]
```diff
@@ -31,6 +31,17 @@ export async function getAllEvents() {
...
 
+/**
+ * Handles the case in which the image URL is not valid
+ * @param {string} url an image url to be added to an event
+ */
+function handleImageURL(url) {
+  if(url !== null) {
+    return url;
+  }
+  return 'https://www.freeiconspng.com/uploads/no-image-icon-11.PNG';
```
[the remark this answers is not in the document]
Comment (IsitaB) [reply src/APIFunctions/Event.js:42]: since you're repeating yourself

[the remark this answers is not in the document]
Comment (IsitaB) [reply src/APIFunctions/Event.js:42]: Also, when an image url is not a real image, 
Screen Shot 2020-08-23 at 9 06 06 PM your hardcoded left padding makes it look a little janky

File: src/Pages/Events/event-page.css [hunk 20]
```diff
@@ -1,55 +1,90 @@
...
 
-.card-row {
-  margin: 0px !important;
+.event-title {
+  font-style: normal;
+  font-weight: bold;
+  font-size: 2vw;
+  padding-left: 5.9vw;
+  color: #000000;
+  justify-content: center;
```
Comment (IsitaB) [inline src/Pages/Events/event-page.css:28]: If you strongly believe in centering, may I suggest making the title's font larger?

File: src/APIFunctions/Event.js (hunk 19 above)
[the remark this answers is not in the document]
Comment (IsitaB) [reply src/APIFunctions/Event.js:42]: and whatever is going on with the spacing above it. Can you handle an invalid image in a better way?

**Comment by quinklan** [comment]
> Screen scaling is important!
> 
> 125% zoom:
> image
> 
> 100% zoom:
> image
> 
> 67% zoom:
> image
> 
> The problem with your code is the hardcoded `px` values in the css file. You generally do not want to use stuff like `min-width: 1000px` because it does not scale well. Instead, look into things like `em`, `rem`, `vw` etc. You want to use _relative lengths_. See [here](https://www.w3schools.com/cssref/css_units.asp) for more units.

I fixed the zooming issue by making the font-size scale with % instead of px.  I believe the only problems left are when a user views the event page on a small screen, such as a phone, or if the user makes the website window tiny. When a user inputs 25 characters consecutively without space in between, the text spills out of the event card as well.

**Comment by jerrylee17** [comment]
image
Simplify the cards (ex: remove image) when the screen is small
